lib/oxrdf/src/blank_node.rs

```diff
@@ -414,4 +466,64 @@ mod tests {
         assert_eq!(align_of::<BlankNode>(), 8);
         assert_eq!(align_of::<BlankNodeRef<'_>>(), 8);
     }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_serde() {
+        let b = BlankNode::new_from_unique_id(0x42);
+        let json = serde_json::to_string(&b).unwrap();
+        assert_eq!(json, "{\"value\":\"42\"}");
+        let b2: BlankNode = serde_json::from_str(&json).unwrap();
+
+        assert_eq!(b2, BlankNode::new("42").unwrap());
+
+        match b2 {
+            BlankNode(BlankNodeContent::Anonymous { str: _, id }) => {
+                assert_eq!(id, 0x42u128.to_ne_bytes());
+            }
+            _ => panic!("Expected anonymous blank node"),
+        }
+
+        let b = BlankNode::new("a").unwrap();
+        let json = serde_json::to_string(&b).unwrap();
+        assert_eq!(json, "{\"value\":\"a\"}");
+        let b2: BlankNode = serde_json::from_str(&json).unwrap();
+        assert_eq!(b2, b);
+
+        let b3: Result<BlankNode, serde_json::Error> =
+            serde_json::from_str::<BlankNode>(&"{\"art\":\"r\", \"value\":\"a\", \"noise\":\"t\"}");
+        assert!(b3.is_err());
+
+        let b4: Result<BlankNode, serde_json::Error> =
+            serde_json::from_str::<BlankNode>(&"{\"art\":\"r\"}");
+        assert!(b4.is_err());
+    }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn as_str_partial_reader() {
```

Comment:
Is this test useful? It looks like a repetition of the previous one

Reply:
Because it uses a `Reader` to deserialise - and `Reader`s require implementors to not borrow strings. See  https://github.com/oxigraph/oxigraph/pull/1272#discussion_r2106251075 for more detail.